docs/metadata.rst

```diff
@@ -153,34 +95,104 @@ explanatory purposes.
               // Contents of the @dev NatSpec field of the state variable
               "details": "Must be set during contract creation. Can then only be changed by the owner"
             }
-          }
-          "events": {
-             "Transfer(address,address,uint256)": {
-               "details": "Emitted when `value` tokens are moved from one account (`from`) toanother (`to`)."
-               "params": {
-                 "from": "The sender address"
-                 "to": "The receiver address"
-                 "value": "The token amount"
-               }
-             }
-          }
+          },
+          // Contents of the @title NatSpec field of the contract
+          "title": "MyERC20: an example ERC20",
+          "version": 1 // NatSpec version
         },
-        // Required: NatSpec user documentation of the contract
+        // Required: NatSpec user documentation of the contract. See "NatSpec Format"
         "userdoc": {
-          "version": 1 // NatSpec version
+          "errors": {
+            "ApprovalCallerNotOwnerNorApproved()": [
+              {
+                "notice": "The caller must own the token or be an approved operator."
+              }
+            ]
+          },
+          "events": {
+            "Transfer(address,address,uint256)": {
+              "notice": "`_value` tokens have been moved from `from` to `to`"
+            }
+          },
           "kind": "user",
           "methods": {
             "transfer(address,uint256)": {
               "notice": "Transfers `_value` tokens to address `_to`"
             }
           },
-          "events": {
-            "Transfer(address,address,uint256)": {
-              "notice": "`_value` tokens have been moved from `from` to `to`"
+          "version": 1 // NatSpec version
+        }
+      },
+      // Required: Compiler settings. Reflects the settings in the JSON input during compilation.
+      // Check the documentation of standard JSON input's "settings" field
+      "settings": {
+        // Required for Solidity: File path and the name of the contract or library this
+        // metadata is created for.
+        "compilationTarget": {
+          "myDirectory/myFile.sol": "MyContract"
+        },
+        // Required for Solidity.
+        "evmVersion": "london"
```

Comment:
Not sure if this is Required but I believe so. Where can I find the schema of the metadata or a validation for that?

Reply:
I have no idea, but the best I can do is the [parser/validator used in our tests](https://github.com/ethereum/solidity/blob/develop/test/Metadata.cpp#L194), which would lead me to believe that `evmVersion` is in fact a mandatory field.